### PR TITLE
ncmdump 1.2.1 (new formula)

### PR DESCRIPTION
**Pull request by taurusxin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/n/ncmdump.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Ncmdump < Formula
+  desc "Convert Netease Cloud Music ncm files to mp3/flac files"
+  homepage "https://github.com/taurusxin/ncmdump"
+  url "https://github.com/taurusxin/ncmdump/archive/refs/tags/1.2.1.tar.gz"
+  sha256 "a1bd97fd1b46f9ba4ffaac0cf6cf1e920b49bf6ec753870ad0e6e07a72c2de2d"
+  license "MIT"
+
```
Comment (p-linnane) [inline Formula/n/ncmdump.rb:7]:
```suggestion
  head "https://github.com/taurusxin/ncmdump.git", branch: "main"

```

File: Formula/n/ncmdump.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  resource("test_input") do
+    url "https://github.com/taurusxin/ncmdump/raw/main/test/test.ncm"
+    sha256 "a1586bbbbad95019eee566411de58a57c3a3bd7c86d97f2c3c82427efce8964b"
+  end
+
+  resource("test_expect") do
+    url "https://github.com/taurusxin/ncmdump/raw/main/test/expect.bin"
+    sha256 "6e0de7017c996718a8931bc3ec8061f27ed73bee10efe6b458c10191a1c2aac2"
+  end
+
```
Comment (p-linnane) [inline Formula/n/ncmdump.rb:19]:
```suggestion

```

File: Formula/n/ncmdump.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      end
+    elsif OS.linux?
+      system "make", "linux"
+    end
+    bin.install "ncmdump"
+  end
+
+  test do
+    resource("test_input").stage testpath
+    resource("test_expect").stage testpath
```
Comment (p-linnane) [inline Formula/n/ncmdump.rb:35]:
```suggestion
    resource "homebrew-test" do
      url "https://raw.githubusercontent.com/taurusxin/ncmdump/516b31ab68f806ef388084add11d9e4b2253f1c7/test/test.ncm"
      sha256 "a1586bbbbad95019eee566411de58a57c3a3bd7c86d97f2c3c82427efce8964b"
    end

    resource "homebrew-expect" do
      url "https://raw.githubusercontent.com/taurusxin/ncmdump/2e40815b5a83236f3feb44720954dd3a02eb00f1/test/expect.bin"
      sha256 "6e0de7017c996718a8931bc3ec8061f27ed73bee10efe6b458c10191a1c2aac2"
    end

    resources.each { |r| r.stage(testpath) }
```

**Review by p-linnane — CHANGES_REQUESTED** [review]
Thanks for your submission. I have a few formatting/cleanup items I'd like addressed prior to merging.

File: Formula/n/ncmdump.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  def install
+    if OS.mac?
+      if Hardware::CPU.arm?
+        system "make", "macos-arm64"
+      elsif Hardware::CPU.intel?
+        system "make", "macos-intel"
+      end
+    elsif OS.linux?
+      system "make", "linux"
+    end
```
Comment (p-linnane) [inline Formula/n/ncmdump.rb:29]:
```suggestion
    os = OS.mac? ? "macos-" : "linux"
    arch = Hardware::CPU.intel? ? "intel" : Hardware::CPU.arch.to_s if OS.mac?
    system "make", "#{os}#{arch}"
```

**Comment by taurusxin** [comment]
> Thanks for your submission. I have a few formatting/cleanup items I'd like addressed prior to merging.

Why there is always a autosquash tag and the commit message is not correct?

**Comment by p-linnane** [comment]
Because there are 2 commits right now. Please squash them into one and it'll be ready.

**Review by p-linnane — APPROVED** [review]

**Pull request merged**